src/main/java/org/folio/inventory/resources/Items.java

```diff
@@ -186,17 +188,14 @@ private void update(RoutingContext routingContext) {
     ItemCollection itemCollection = storage.getItemCollection(context);
     UserCollection userCollection = storage.getUserCollection(context);
 
-    itemCollection.findById(routingContext.request().getParam("id"), getItemResult -> {
+    final String itemId = routingContext.request().getParam("id");
+    itemCollection.findById(itemId, getItemResult -> {
       Item oldItem = getItemResult.getResult();
       if (oldItem != null) {
-        if (!Objects.equals(newItem.getHrid(), oldItem.getHrid())) {
-          log.warn("The HRID property can not be updated, old value is '{}' but new is '{}'",
-            oldItem.getHrid(), newItem.getHrid()
-          );
-
-          JsonResponse.unprocessableEntity(routingContext.response(),
-            "HRID can not be updated", "hrid", newItem.getHrid()
-          );
+        Optional<ValidationError> updateValidationErrors =
+          validateItemUpdate(oldItem, newItem);
```

Comment:
Has `Optional` been used in other places to detect the presence of errors or not?

Reply:
No, I have not seen similar approach. Actually, there is no established approach for validation within mod-inventory.

Reply:
Are you aware of one in other modules that EPAM folks are working on?

(The `Result` based approach in mod-circulation appears to be relatively unusual)

Reply:
Some of modules uses Future-based validation, it is similar to `Result` approach but validation failures/record not found/etc is a sub-class of an exception.

Reply:
Yeah, I was trying to avoid `exceptions` with `Result`, it didn't quite work out that way. I was also trying to keep asynchronicity separate as well, with mixed results.